src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -158,6 +158,15 @@ public function buildForm(FormBuilderInterface $builder, array $options)
      */
     public function buildView(FormView $view, FormInterface $form, array $options)
     {
+        $choiceTranslationDomain = $options['choice_translation_domain'];
+        $translationDomain = $options['translation_domain'];
+
+        if ($view->parent) {
+            if (null === $choiceTranslationDomain) {
+                $choiceTranslationDomain = $view->parent->vars['translation_domain'];
```

Comment:
I'm not saying it should be removed entirely. I'm sayign the current implementation is buggy.

Try to set the ``choice_translation_domain`` explicitly with a non-inherited translation domain.

you should not inherit the choice_translation_domain from the parent translation_domain. you should inherit it from the *current* translation domain (be it inherited or no)